src/types/int_type.rs

```diff
@@ -58,14 +57,7 @@ impl StringRadix {
         }
 
         // and all digits must be in the radix' character set
-        let mut it = slice.chars();
-        match self {
-            StringRadix::Binary => it.all(|c| matches!(c, '0'..='1')),
-            StringRadix::Octal => it.all(|c| matches!(c, '0'..='7')),
-            StringRadix::Decimal => it.all(|c| matches!(c, '0'..='9')),
-            StringRadix::Hexadecimal => it.all(|c| matches!(c, '0'..='9' | 'a'..='f' | 'A'..='F')),
-            StringRadix::Alphanumeric => it.all(|c| matches!(c, '0'..='9' | 'a'..='z' | 'A'..='Z')),
-        }
+        slice.chars().all(|c| c.is_digit(*self as u32))
```

Comment:
I don't think this is correct since hex and alpha can have numerics

Reply:
Oh interesting, I didn't know that. But what about subsets? For example, binary should only be 0 or 1, and not 2-9, a-z, A-Z, etc?

Reply:
Yes, that's what the radix argument is for :)

Reply:
ohhhh haha